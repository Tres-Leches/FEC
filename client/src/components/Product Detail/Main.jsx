import React from 'react';
import axios from 'axios';
import './productDetail.css';

import Add2Cart from './Add2Cart';
import DefaultView from './DefaultView';
import ExpandedView from './ExpandedView';
import ProductInfo from './ProductInfo';
import StyleSelector from './StyleSelector';
import Share from './Share';
import EndInfo from './EndInfo';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      product_ids: [16392, 16465, 16056, 16084, 16154, 16060, 16072, 16073],
      product_id: 16060,
      zoomed: false,
      product: null,
      styles: [],
      style: null,
      mainPhoto: null,
    };
  }

  componentDidMount() {
    axios.get(`/api/products/${this.state.product_id}/styles`)
      .then((response) => {
        this.setState({styles: response.data.results, style: response.data.results[0], mainPhoto: response.data.results[0].photos[0]})
      })
      .catch(err => console.error(err))
    axios.get(`/api/products/${this.state.product_id}`)
      .then((response) => {
        this.setState({product: response.data})
      })
      .catch(err => console.error(err))
  }

  changeStyle(style) {
    this.setState({style})
  }

  changeMainPhoto (photo) {
    this.setState({mainPhoto: photo})
  }

  changeView() {
    this.setState({zoomed: !this.state.zoomed})

  }

  render() {
    if(this.state.style && this.state.product){
      let expand;
      if (this.state.zoomed) {
        expand = <ExpandedView style={this.state.style}
          mainPhoto={this.state.mainPhoto}
          changeMainPhoto={this.changeMainPhoto.bind(this)}
          changeView={this.changeView.bind(this)}/> ;
        document.body.style.overflow = 'hidden';
      } else {
        expand = "";
        document.body.style.overflow = 'visible';
      }
      return (
        <div>
          {expand}
          <div className="overview">

            <div className="view">
              <DefaultView style={this.state.style} changeView={this.changeView.bind(this)} mainPhoto={this.state.mainPhoto} changeMainPhoto={this.changeMainPhoto.bind(this)}/>
              <div className="info">
              <ProductInfo product={this.state.product} productId={this.state.product_id}/>
              <StyleSelector styles={this.state.styles} style={this.state.style} changeStyle={this.changeStyle.bind(this)}/>
              <Add2Cart style={this.state.style}/>
              <Share />
            </div>
            </div>

            <div className="viewLastInfo">
              <EndInfo product={this.state.product}/>
            </div>
          </div>
        </div>
      );
    } else {
      return (
        <div></div>
      )
    }
  }
}

export default Main;
