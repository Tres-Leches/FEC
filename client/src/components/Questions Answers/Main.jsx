/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';
import _ from 'underscore';
import SearchBar from './SearchBar';
import QuestionsList from './QuestionsList';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      questions: [],
      filteredQuestions: [],
      searchQuery: '',
    };
    this.getQuestions = this.getQuestions.bind(this);
    this.setSearchQuery = this.setSearchQuery.bind(this);
    this.filterQuestions = this.filterQuestions.bind(this);
  }

  componentDidMount() {
    this.getQuestions();
  }

  setSearchQuery(e) {
    this.setState({ searchQuery: e.target.value }, () => {
      this.filterQuestions();
    });
  }

  getQuestions() {
    const productId = 16392; // Placeholder
    axios.get(`/api/qa/questions/${productId}`)
      .then((response) => {
        const questions = response.data.results;
        questions.sort((a, b) => (
          b.question_helpfulness - a.question_helpfulness
        ));
        this.setState({ questions });
      })
      .then(() => { this.filterQuestions(); })
      .catch((err) => console.error(err));
  }

  filterQuestions() {
    const { searchQuery, questions } = this.state;
    let filtered = [];
    if (searchQuery.length >= 3) {
      questions.forEach((question) => {
        const query = searchQuery.toLowerCase();
        const qBody = question.question_body.toLowerCase();
        if (qBody.includes(query)) {
          filtered.push(question);
        } else {
          // Else search through answers of each question_id.
          // If answer.body, aBody, includes query, return question
          const { answers } = question;
          Object.keys(answers).forEach((answerId) => {
            const aBody = question.answers[`${answerId}`].body.toLowerCase();
            if (aBody.includes(query)) {
              if (_.findWhere(filtered, aBody) == null) {
                filtered.push(question);
              }
            }
          });
        }
      });
    } else {
      filtered = questions;
    }
    this.setState({ filteredQuestions: filtered });
  }

  render() {
    const { filteredQuestions, searchQuery } = this.state;

    return (
      <div>
        Hello from Questions and Answers
        <SearchBar
          searchQuery={searchQuery}
          setSearchQuery={this.setSearchQuery}
          handleSearchSubmit={this.handleSearchSubmit}
        />
        <QuestionsList questions={filteredQuestions} />
      </div>
    );
  }
}

export default Main;
