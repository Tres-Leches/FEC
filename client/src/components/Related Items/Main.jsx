import React from 'react';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    return (
      <div>
        Hello from Related Items
      </div>
    );
  }
}

export default Main;
