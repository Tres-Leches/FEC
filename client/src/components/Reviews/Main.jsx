/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';
import ReviewList from './ReviewList.jsx';
import MetaData from './MetaData.jsx';
import './reviews.css';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      productId: this.props.productId,
      product: this.props.product,
      reviews: '',
      metaData: '',
    };
  }

  componentDidMount() {
    Promise.all([
      axios.get(`/api/reviews/${this.state.productId}/relevant`),
      axios.get(`/api/reviews/meta/${this.state.productId}`),
    ]).then(([res1, res2]) => {
      this.setState({
        reviews: res1.data,
        metaData: res2.data,
      }, () => console.log(this.state));
    });
  }

  getReviews(sort) {
    axios.get(`/api/reviews/${this.state.productId}/${sort}`)
      .then((res) => {
        this.setState({
          reviews: res.data,
        });
      }).then(() => console.log(this.state)).catch((err) => (console.log(err)));
  }

  render() {
    if (!this.state.reviews || !this.state.metaData) {
      return (
        <div />
      );
    }
    return (
      <div>
        <div>Ratings &amp; Reviews</div>
        <MetaData metaData={this.state.metaData} />
        <ReviewList
          reviews={this.state.reviews}
          meta={this.state.metaData}
          getReviews={this.getReviews.bind(this)}
        />
      </div>
    );
  }
}

export default Main;
