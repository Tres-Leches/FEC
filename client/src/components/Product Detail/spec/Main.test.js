import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Main from '../Main.jsx';
import 'font-awesome/css/font-awesome.min.css';

describe('Main', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Main debug />);
    expect(component).toMatchSnapshot();
  });


});
