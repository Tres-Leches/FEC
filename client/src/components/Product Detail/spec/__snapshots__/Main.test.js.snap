// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add2Cart should render correctly with props 1`] = `ShallowWrapper {}`;

exports[`DefaultView should render correctly 1`] = `ReactWrapper {}`;

exports[`DefaultView should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`DefaultView should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`DefaultView should render correctly with props 1`] = `ShallowWrapper {}`;

exports[`EndInfo should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`EndInfo should render correctly with props 1`] = `ShallowWrapper {}`;

exports[`ExpandedView should render correctly with props 1`] = `ShallowWrapper {}`;

exports[`Main should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Main should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`ProductInfo should render correctly 1`] = `ReactWrapper {}`;

exports[`ProductInfo should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`ProductInfo should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`ProductInfo should render correctly with props 1`] = `ShallowWrapper {}`;

exports[`Share should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Share should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Share should render correctly without any props 1`] = `ShallowWrapper {}`;

exports[`StyleSelector should render correctly with props 1`] = `ShallowWrapper {}`;
